[≥ 1.32.0] TS2344: Type '"event" | "page" | "userProperties"' does not satisfy the constraint 'keyof TargetingParameters'

> [!NOTE]
> this analysis was AI-generated based on a typing error in our CI

## Summary

Type definitions in `@amplitude/session-replay-browser` reference `'page'` as a key of `TargetingParameters` from `@amplitude/targeting`, but the bundled `@amplitude/targeting@0.2.0` dependency does not include `page` in its `TargetingParameters` interface. This causes a `TS2344` compile error in any project with `skipLibCheck: false`.

## Affected versions

| Package | Version |
|---|---|
| `@amplitude/session-replay-browser` | `1.32.0` – `1.34.2` (latest) |
| `@amplitude/targeting` | `0.2.0` |

**Last working version:** `@amplitude/session-replay-browser@1.31.7`

## Error

```
../../node_modules/@amplitude/session-replay-browser/lib/esm/typings/session-replay.d.ts:83:78
  error TS2344: Type '"event" | "page" | "userProperties"' does not satisfy the constraint 'keyof TargetingParameters'.
    Type '"page"' is not assignable to type 'keyof TargetingParameters'.

83      evaluateTargetingAndCapture: (targetingParams: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>, isInit?: boolean, forceRestart?: boolean, forceTargetingReevaluation?: boolean) => Promise<void>;
```

## Root cause

`session-replay-browser` ≥ 1.32.0 added `'page'` to the `Pick<>` in `evaluateTargetingAndCapture`:

```ts
// lib/esm/typings/session-replay.d.ts (>= 1.32.0)
evaluateTargetingAndCapture: (
  targetingParams: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>,
  ...
) => Promise<void>;
```

However, `@amplitude/targeting@0.2.0` — which this package declares as its own dependency — defines `TargetingParameters` without a `page` key:

```ts
// @amplitude/targeting@0.2.0 — lib/esm/typings/targeting.d.ts
export interface TargetingParameters {
  event?: Event;
  userProperties?: { [key: string]: any };
  deviceId?: string;
  flag: EvaluationFlag;
  sessionId: number;
  apiKey: string;
  loggerProvider: Logger;
  // ← no `page` key
}
```

The session-replay types were updated to reference `page` without a corresponding update to `@amplitude/targeting`.

## Suggested fix

Either:

1. Add a `page` property to `TargetingParameters` in `@amplitude/targeting` and release a new version, **or**
2. Remove `'page'` from the `Pick<>` in `session-replay.d.ts` to match the actual `@amplitude/targeting@0.2.0` interface.

--- a/packages/session-replay-browser/src/session-replay.ts
+++ b/packages/session-replay-browser/src/session-replay.ts
@@ -15,7 +15,6 @@ import {
 
 // Import only specific types to avoid pulling in the entire rrweb-types package
 import { eventWithTime, EventType as RRWebEventType, scrollCallback } from '@amplitude/rrweb-types';
-import { TargetingParameters } from '@amplitude/targeting';
 import { createSessionReplayJoinedConfigGenerator } from './config/joined-config';
 import {
   LoggingConfig,
@@ -54,6 +53,7 @@ import {
   EventType,
   SessionIdentifiers as ISessionIdentifiers,
   SessionReplayOptions,
+  SessionReplayTargetingInput,
 } from './typings/session-replay';
 import { VERSION } from './version';
 
@@ -75,7 +75,7 @@ export class SessionReplay implements AmplitudeSessionReplay {
   eventCount = 0;
   eventCompressor: EventCompressor | undefined;
   sessionTargetingMatch = false;
-  private lastTargetingParams?: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>;
+  private lastTargetingParams?: SessionReplayTargetingInput;
   private lastShouldRecordDecision?: boolean;
 
   // Visible for testing only
@@ -159,7 +159,7 @@ export class SessionReplay implements AmplitudeSessionReplay {
     };
   }
 
-  private getCurrentPageForTargeting(): Pick<TargetingParameters, 'page'>['page'] {
+  private getCurrentPageForTargeting(): SessionReplayTargetingInput['page'] {
     const currentUrl = getGlobalScope()?.location?.href;
     return currentUrl != null ? { url: currentUrl } : undefined;
   }
@@ -384,7 +384,7 @@ export class SessionReplay implements AmplitudeSessionReplay {
   };
 
   evaluateTargetingAndCapture = async (
-    targetingParams: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>,
+    targetingParams: SessionReplayTargetingInput,
     isInit = false,
     forceRestart = false,
     forceTargetingReevaluation = false,

--- a/packages/session-replay-browser/src/targeting/targeting-manager.ts
+++ b/packages/session-replay-browser/src/targeting/targeting-manager.ts
@@ -1,6 +1,7 @@
 import type { TargetingParameters } from '@amplitude/targeting';
 import { TargetingConfig } from '../config/types';
 import { Logger } from '@amplitude/analytics-types';
+import { SessionReplayTargetingInput } from '../typings/session-replay';
 import { targetingIDBStore } from './targeting-idb-store';
 
 export const evaluateTargetingAndStore = async ({
@@ -15,7 +16,7 @@ export const evaluateTargetingAndStore = async ({
   targetingConfig: TargetingConfig;
   loggerProvider: Logger;
   apiKey: string;
-  targetingParams?: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>;
+  targetingParams?: SessionReplayTargetingInput;
   urlChange?: boolean;
 }) => {
   await targetingIDBStore.clearStoreOfOldSessions({

--- a/packages/session-replay-browser/src/typings/session-replay.ts
+++ b/packages/session-replay-browser/src/typings/session-replay.ts
@@ -1,6 +1,12 @@
 import { AmplitudeReturn, ServerZone } from '@amplitude/analytics-core';
-import type { TargetingParameters } from '@amplitude/targeting';
+import type { Event } from '@amplitude/analytics-types';
 import { SessionReplayJoinedConfig, SessionReplayLocalConfig, SessionReplayVersion } from '../config/types';
+
+export interface SessionReplayTargetingInput {
+  event?: Event;
+  userProperties?: { [key: string]: any };
+  page?: { url?: string };
+}
 
 export type StorageData = {
   totalStorageSize: number;
@@ -95,7 +101,7 @@ export interface AmplitudeSessionReplay {
   getSessionId: () => string | number | undefined;
   getSessionReplayProperties: () => { [key: string]: boolean | string | null };
   evaluateTargetingAndCapture: (
-    targetingParams: Pick<TargetingParameters, 'event' | 'userProperties' | 'page'>,
+    targetingParams: SessionReplayTargetingInput,
     isInit?: boolean,
     forceRestart?: boolean,
     forceTargetingReevaluation?: boolean,

